### DISCUSSION TITLE
Expose `sendMessagePromise`

### DISCUSSION THREAD

**Discussion by Baael** [shannonhochkins/ha-component-kit#68] [Ideas]
First of all - nice project!
 
My idea is to expose `sendMessagePromise` in `useHass` hook. 
Why? 
I am right now creating map for polygon zones, and it is much faster, lighter and easier to work with messages.
For example:
```js
await connection.sendMessagePromise({ type: "zone/list" })
await connection.sendMessagePromise({ type: "zone/create", ... })
await connection.sendMessagePromise({ type: "zone/delete", ... })
await connection.sendMessagePromise({ type: "zone/update", ... })
```

My idea is to solve it that way:

```js
const { sendMessage } = useHass()
```

but I think it would be also nice (because why not):
```js
const { list, create, delete, update } = useZones()
```

You can check the problem with zones here: https://community.home-assistant.io/t/wth-we-only-have-round-shaped-zones/468137/12

Basically in FE idea is to add map with free version of https://geoman.io/ and allow users to add polygon zones.

**Comment by shannonhochkins** [comment]
Good idea! Although you should already have access to sendMessagePromise from the connection object 

I've tried to build hakit in a somewhat generic way, yet still providing anything users need to go nuts!

```
const { useStore } = useHass();
const connection = useStore(store => store.connection);
// now you can use
// connection.sendMessagePromise :) 
```

If you come up with something that's going to be useful for others, please do share it and I'll integrate it properly 

Have fun!

**Reply by shannonhochkins** [reply]
If you're building a client side map, with hakit, please share it I'd love to add it as a card

**Reply by Baael** [reply]
Sure! When I will be far enough.
Right now I was concentrated more on fixing python part in overriding zone component and making everything work correctly.
Editor is on a horizon: https://clipchamp.com/watch/K2n4uqWcGax

**Reply by shannonhochkins** [reply]
Awesome!
